[GTK] Remove WebKitPrintCustomWidget from modern API

mcatanzaro opened the pull request:
#### e3ebba80adf1cc861a986732fe154aa72731b2b1
<pre>
[GTK] Remove WebKitPrintCustomWidget from modern API
<a href="https://bugs.webkit.org/show_bug.cgi?id=244513">https://bugs.webkit.org/show_bug.cgi?id=244513</a>

Reviewed by Carlos Garcia Campos.

Attaching custom widgets to the print dialog just cannot work when the
print dialog is out of process, and we really need to move to an out of
process dialog to fix printing in Flatpak.

Now, it would be possible to keep this functionality for non-flatpak
apps and leave it broken in Flatpak, but I&apos;d much rather reduce user
experience divergence.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformGTKDeprecated.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesGTKDeprecated.txt:
* Source/WebKit/UIProcess/API/glib/WebKitAutocleanups.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintCustomWidget.cpp:
(webkit_print_custom_widget_class_init):
* Source/WebKit/UIProcess/API/gtk/WebKitPrintCustomWidget.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
(webkit_print_operation_class_init):
(webkitPrintOperationRunDialog):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestPrinting.cpp:
(beforeAll):

Canonical link: <a href="https://commits.webkit.org/259286@main">https://commits.webkit.org/259286@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a23cad2839ba6201b8eef34348982732f1f06dde

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104161 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13253 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37080 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113371 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173661 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108099 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14320 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4168 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96389 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112432 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109931 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11029 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94094 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38698 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92882 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25706 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80356 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6617 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27072 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6753 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3616 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12762 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46618 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6403 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8535 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

carlosgcampos reviewed (approved):
Ok, I don't even know if this is actually used.

mcatanzaro committed:
[GTK] Remove WebKitPrintCustomWidget from modern API
https://bugs.webkit.org/show_bug.cgi?id=244513

Reviewed by Carlos Garcia Campos.

Attaching custom widgets to the print dialog just cannot work when the
print dialog is out of process, and we really need to move to an out of
process dialog to fix printing in Flatpak.

Now, it would be possible to keep this functionality for non-flatpak
apps and leave it broken in Flatpak, but I'd much rather reduce user
experience divergence.

* Source/WebKit/PlatformGTK.cmake:
* Source/WebKit/PlatformGTKDeprecated.cmake:
* Source/WebKit/SourcesGTK.txt:
* Source/WebKit/SourcesGTKDeprecated.txt:
* Source/WebKit/UIProcess/API/glib/WebKitAutocleanups.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintCustomWidget.cpp:
(webkit_print_custom_widget_class_init):
* Source/WebKit/UIProcess/API/gtk/WebKitPrintCustomWidget.h.in:
* Source/WebKit/UIProcess/API/gtk/WebKitPrintOperation.cpp:
(webkit_print_operation_class_init):
(webkitPrintOperationRunDialog):
* Tools/TestWebKitAPI/Tests/WebKitGtk/TestPrinting.cpp:
(beforeAll):

Canonical link: https://commits.webkit.org/259286@main
